mibidata/tiff.py

```diff
@@ -93,27 +93,13 @@ def write(filename, image, sed=None, optical=None, ranges=None,
                   image.data.shape[1] * 1e4 / float(image.size),
                   'cm')
 
-    metadata = {
-        'mibi.run': getattr(image, 'run'),
-        'mibi.version': getattr(image, 'version'),
-        'mibi.instrument': getattr(image, 'instrument') or 'MIBI',
-        'mibi.slide': getattr(image, 'slide'),
-        'mibi.dwell': getattr(image, 'dwell'),
-        'mibi.scans': getattr(image, 'scans'),
-        'mibi.aperture': getattr(image, 'aperture'),
-        'mibi.description': getattr(image, 'point_name'),
-        'mibi.folder': getattr(image, 'folder'),
-        'mibi.tissue': getattr(image, 'tissue'),
-        'mibi.panel': getattr(image, 'panel'),
-        'mibi.mass_offset': getattr(image, 'mass_offset'),
-        'mibi.mass_gain': getattr(image, 'mass_gain'),
-        'mibi.time_resolution': getattr(image, 'time_resolution'),
-        'mibi.miscalibrated': getattr(image, 'miscalibrated'),
-        'mibi.check_reg': getattr(image, 'check_reg'),
-        'mibi.filename': getattr(image, 'filename'),
-    }
+    metadata = {'mibi.' + key: value for key, value in \
+                image.metadata().items() \
+                if key not in ['date']} # datetime objects are not
+                                        # JSON serializable
+
```

Comment:
The original idea here was the namespace the required attributes with the `'mibi.'` prefix, but now we are adding that to both them and user-defined extras. I'm not with the name-spacing at all at this point, but given that we've been using it maybe it's better we stick to that plan and don't add it to arbitrary attributes? Unfortunately we are not taking advantage of this in terms of how we convert TIFFs to/from MibiImages at this point but we can address that later.

Reply:
Ok, but I need to tag the user-defined metadata somehow, otherwise it's very hard to locate them in the tiff, when reading a tiff file and retrieving the metadata attributes for the MibiImage. I'm using now `user-def.` to tag the user-defined attributes, in order not to mix the attributes with the required ones, marked as `mibi.`.

Reply:
Can't you grab everything else from the dict that doesn't have the prefix / isn't listed as required? Similar to how you already construct a MibiImage from kwargs?

Reply:
I will have a look, but I think there might be more info stored there.

Reply:
OK I implemented the latter: I add a list of user-defined attribute keywords to the TIFF metadata, that makes the retrieval of the user-defined attributes straightforward, without the need for a special namespace prefix, similar to what I implemented for MibiImage.